### ISSUE TITLE
Question: Integration in Seurat workflow?

### ISSUE THREAD

**Issue by tilofrei** [mohuangx/SAVER#18]
Dear SAVER-Team,

thank you for providing this great package. I am wondering what is the usual integration in the Seurat workflow and especially if I still need `Seurat::ScaleData()`  and on which object I run the imputation (`data@data`, `data@raw.data`, `data@scale.data`).

My simplified integration would be like that:

```
# 1
data = read.table("test.txt")
data = CreateSeuratObject()
data <- NormalizeData()
data <- ScaleData()

# 2
data@imputed = saver(data@data, size.factor=1)

# 3: Clustering, plotting etc
```

Many thanks!

**Comment by mohuangx** [comment]
I would run SAVER on the raw data first and then use the SAVER results as the input into the Seurat object. For example:

```
data <- as.matrix(read.table("test.txt"))
data.saver <- saver(data, estimates.only = TRUE)
seurat.obj <- CreateSeuratObject(raw.data = data.saver)
seurat.obj <- NormalizeData(seurat.obj)
seurat.obj <- ScaleData(seurat.obj)
```

**Comment by tilofrei** [comment]
Thank you for your quick response!

I also found a very recent discussion on that topic in the biostars-forum that recommends running Seurat on the raw data and only imputate before plotting the data (but they don't mention the scaling):
https://www.biostars.org/p/338852/

Any thoughts on that?

**Comment by mohuangx** [comment]
There are some good points raised in the discussion regarding the need for caution when using imputed data in downstream analysis. However, we have found that SAVER is quite robust to introducing artificial bias in the data. In our paper, we performed imputation prior to running the Seurat workflow and was able to produce biologically meaningful clusters. 

One of the benefits we found in running Seurat on the imputed data was better identification of highly variable genes and more signal in PCA and clustering. Your mileage may vary but it wouldn't be too much additional effort to see what kind of results you get by running Seurat on the imputed data as well.

If you want to work with the raw data but still add the SAVER results to the `imputed` slot, I would run the following:
```
data <- as.matrix(read.table("test.txt"))
data.saver <- saver(data, estimates.only = TRUE)
seurat.obj <- CreateSeuratObject(raw.data = data)
seurat.obj <- NormalizeData(seurat.obj)
seurat.obj <- ScaleData(seurat.obj)
seurat.obj@imputed <- data.saver
```
